### PR TITLE
Poincare refactoring

### PR DESCRIPTION
**Pull request by EmilieNumworks**
Before:
```
========= BUILD OUTPUT ========
File:  epsilon.elf
Code:  420076 bytes
Data:  247988 bytes
Total: 652 kB (668064 bytes)
===============================
```

After:
```
========= BUILD OUTPUT ========
File:  epsilon.elf
Code:  428532 bytes
Data:  248060 bytes
Total: 660 kB (676592 bytes)
===============================
```

### CODE REVIEW

File: apps/calculation/calculation.cpp [hunk 1]
```diff
@@ -186,24 +191,24 @@ bool Calculation::shouldDisplayApproximateOutput(Context * context) {
   if (strcmp(m_exactOutputText, m_approximateOutputText) == 0) {
```
Comment (Ecco) [inline apps/calculation/calculation.cpp]: Rename to `shouldOnlyDisplayApproximateOutput`

File: poincare/include/poincare/expression.h [hunk 2]
```diff
@@ -290,8 +287,10 @@ class Expression {
     return nullptr;
   }
   /* Evaluation Engine */
-  virtual Expression * privateApproximate(SinglePrecision p, Context& context, AngleUnit angleUnit) const = 0;
-  virtual Expression * privateApproximate(DoublePrecision p, Context& context, AngleUnit angleUnit) const = 0;
+  template<typename T> static Expression * CreateDecimal(T f);
+  template<typename T> static Expression * complexToExpression(std::complex<T> c, ComplexFormat complexFormat);
```
Comment (Ecco) [inline poincare/include/poincare/expression.h]: Make this a member function on Evaluation<T>

File: poincare/src/evaluation.cpp [hunk 3]
```diff
@@ -0,0 +1,261 @@
...
+    result.imag(0);
+  }
+  if (c.imag() == 0 && d.imag() == 0 && (c.real() >= 0 || d.real() == std::round(d.real()))) {
+    result.imag(0);
+  }
+  return result;
+}
+
+template<typename T>
+std::complex<T> Complex<T>::approximate(const std::complex<T>& c, ComplexFunction<T> function) {
```
Comment (Ecco) [inline poincare/src/evaluation.cpp]: * Remove `approximate` from `Evaluation`
* Remove the `sin`, `cos` et al from `Evaluation` as well
* In Sine:
```
static std::complex<T> Sine::computeOnComplex(std::complex<T> c) {
  std::complex res = std::sin(c);
  return Trigonometry::roundToMeaningfulDigits(res);
}
```
* In Power:
```
static std::complex<T> Sine::computeOnComplex(std::complex<T> c) {
 // Here, the code that was in `Evaluation`
}
```

**Comment by EmilieNumworks** [comment]
Change 'PrintFloat::convertFloatToText' to use the same meaning of 'significant digits' as the 'Decimal::convertToText'.

**Pull request closed without merging**